e2e/playwright/search/src/tests/search-filters-logic.spec.ts

```diff
@@ -0,0 +1,125 @@
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { expect } from '@playwright/test';
+import { ApiClientFactory, Utils, test, NodesApi, TrashcanApi, FileActionsApi } from '@alfresco/playwright-shared';
+
+test.describe('Search - Filters - Logic', () => {
+  let nodesApi: NodesApi;
+  let trashcanApi: TrashcanApi;
+  let fileActionsApi: FileActionsApi;
+
+  const randomUsername = Utils.random();
+  const randomFilename = Utils.random();
+  const username = `user1+${randomUsername}`;
+  const logicFile1 = `${username}-${randomFilename}logic-${randomFilename}cats-${randomFilename}clouds-${randomFilename}`;
+  const logicFile1Split = logicFile1.split('-');
+  const logicFile2 = `${username}-${randomFilename}logic-${randomFilename}cats-${randomFilename}sky-${randomFilename}`;
+
+  test.beforeEach(async ({ loginPage, searchPage }) => {
+    await Utils.tryLoginUser(loginPage, username, username, 'beforeEach failed');
+    await searchPage.navigate();
+  });
+
+  test.beforeAll(async () => {
+    try {
+      const apiClientFactory = new ApiClientFactory();
+      await apiClientFactory.setUpAcaBackend('admin');
+      await apiClientFactory.createUser({ username });
+      nodesApi = await NodesApi.initialize(username, username);
+      trashcanApi = await TrashcanApi.initialize(username, username);
+      fileActionsApi = await FileActionsApi.initialize(username, username);
+      await nodesApi.createFile(logicFile1);
+      await nodesApi.createFile(logicFile2);
+      await fileActionsApi.waitForNodes(logicFile1, { expect: 1 });
+      await fileActionsApi.waitForNodes(logicFile2, { expect: 1 });
+    } catch (error) {
+      console.error(`beforeAll failed: ${error}`);
+    }
+  });
+
+  test.afterAll(async () => {
+    await Utils.deleteNodesSitesEmptyTrashcan(nodesApi, trashcanApi, 'afterAll failed');
+  });
+
+  test('[C699500] Filter with Match All', async ({ searchPage }) => {
+    await searchPage.searchFilters.logicFilter.click();
+    await searchPage.searchFiltersLogic.matchAllInput.fill(
+      `${logicFile1Split[0]} ${logicFile1Split[1]} ${logicFile1Split[2]} ${logicFile1Split[3]} ${logicFile1Split[4]}`
+    );
+    await searchPage.searchFiltersLogic.applyButton.click();
+    await searchPage.dataTable.progressBarWaitForReload();
+
+    await expect(searchPage.dataTable.getRowByName(logicFile1)).toBeVisible();
+    await expect(searchPage.dataTable.getRowByName(logicFile2)).not.toBeVisible();
+  });
+
+  test('[C699501] Filter with Match Any', async ({ searchPage }) => {
+    await searchPage.searchFilters.logicFilter.click();
+    await searchPage.searchFiltersLogic.matchAnyInput.fill(`${logicFile1Split[1]}-${logicFile1Split[2]}-${logicFile1Split[3]} ${logicFile1Split[4]}`);
+    await searchPage.searchFiltersLogic.applyButton.click();
+    await searchPage.dataTable.progressBarWaitForReload();
+
+    expect(await searchPage.dataTable.getRowsCount()).toBe(3);
```

Comment:
Is it possible to edit getRowsCount() method to not count the first row with column names? It would be easier to understand tests in the future. Right now the logic says that there should be 3 items in the dataTable, but in reality there are only 2 which is a little misleading 😄

Reply:
I was afraid to do so because it's widely used method :D I can try

Reply:
I will try in separate PR as it will require changes in many files